From b41c7d9ae49b91a1d5760f707c0347c4732e78a3 Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsh.thakkar@xilinx.com>
Date: Wed, 7 Aug 2019 11:44:34 -0700
Subject: [PATCH 0655/1851] v4l: xilinx: dma: Don't start DMA in STREAM_ON for
 low latency capture

commit f004ff3e7396e59106c3a94c68604c062da43fe3 from
https://github.com/Xilinx/linux-xlnx.git

This patch ensures that the DMA engine is not started for low latency
capture use cases. Applications will start the DMA using S_CTRL at
later point of time when it is needed.

This patch also disables the auto resatrt mode in dma operation for
low latency capture.

Signed-off-by: Devarsh Thakkar <devarsh.thakkar@xilinx.com>
Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index dece219872b3..8299cda085d6 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -676,8 +676,11 @@ static int xvip_dma_start_streaming(struct vb2_queue *vq, unsigned int count)
 
 	/* Start the DMA engine. This must be done before starting the blocks
 	 * in the pipeline to avoid DMA synchronization issues.
+	 * We dont't want to start DMA in case of low latency capture mode,
+	 * applications will start DMA using S_CTRL at later point of time.
 	 */
-	dma_async_issue_pending(dma->dma);
+	if (!dma->low_latency_cap)
+		dma_async_issue_pending(dma->dma);
 
 	/* Start the pipeline. */
 	ret = xvip_pipeline_set_stream(pipe, true);
-- 
2.31.1

